File: lib/features/dashboard/domain/models/month_summary.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'month_summary.freezed.dart';
part 'month_summary.g.dart';

@freezed
abstract class MonthSummary with _$MonthSummary {
  const factory MonthSummary({
    required String yearMonth, // "2025-03"
    @Default(0.0) double totalIncome,
    @Default(0.0) double totalExpense,
    @Default(0.0) double totalSavings,
    @Default(0.0) double netBalance,
    @Default(0.0) double carryOver,
    @Default(0.0) double netWithCarryOver,
    @Default(0.0) double savingsRate,
    @Default(0.0) double expenseRate,
    @Default(0) int healthScore,
    required DateTime updatedAt,
  }) = _MonthSummary;

  factory MonthSummary.fromJson(Map<String, dynamic> json) =>
      _$MonthSummaryFromJson(json);
}

File: SavvyApp/Packages/Core/SavvyFoundation/Sources/SavvyFoundation/Models/MonthSummary.swift
import Foundation

public struct MonthSummary: Codable, Identifiable, Hashable, Sendable {
    public var id: String { yearMonth }

    public let yearMonth: String
    public let totalIncome: Decimal
    public let totalExpense: Decimal
    public let totalSavings: Decimal
    public let netBalance: Decimal
    public let carryOver: Decimal
    public let netWithCarryOver: Decimal
    public let savingsRate: Double
    public let expenseRate: Double
    public let healthScore: Int
    public let updatedAt: Date

    public init(
        yearMonth: String,
        totalIncome: Decimal,
        totalExpense: Decimal,
        totalSavings: Decimal,
        netBalance: Decimal,
        carryOver: Decimal,
        netWithCarryOver: Decimal,
        savingsRate: Double,
        expenseRate: Double,
        healthScore: Int,
        updatedAt: Date = Date()
    ) {
        self.yearMonth = yearMonth
        self.totalIncome = totalIncome
        self.totalExpense = totalExpense
        self.totalSavings = totalSavings
        self.netBalance = netBalance
        self.carryOver = carryOver
        self.netWithCarryOver = netWithCarryOver
        self.savingsRate = savingsRate
        self.expenseRate = expenseRate
        self.healthScore = healthScore
        self.updatedAt = updatedAt
    }
}
